### PR TITLE
Test factories

### LINKED ISSUE #614: Use factories for unit tests
The unit tests have hard-coded dependencies against the test data. This means that we need to build a new test data set whenever the underlying data change, and makes it hard to generate test cases that meet specific requirements (e.g. unicode in names). Tests should use data from factories instead of production data for greater flexibility.

### PR DESCRIPTION
**Pull request by jmcarp**
Refactor tests to use factories rather than real data.

* Use tables rather than materialized views for testing. Otherwise, we can't insert fake data, since views are read-only
* Wrap each test in a transaction and rollback on `tearDown`
* Clear test database after each test suite
* Use separate database for testing; defaults to `cfdm-unit-test` but can be controlled with the `SQLA_TEST_CONN` environment variable

[Resolves #614]

### CODE REVIEW

File: webservices/resources/reports.py [hunk 1]
```diff
@@ -531,5 +532,6 @@ def get_reports(self, committee_id, reports_class, args, page_num, per_page):
             reports = reports.filter(reports_class.cycle.in_(args['cycle'].split(',')))
 
         count = reports.count()
+        import ipdb; ipdb.set_trace()
```
Comment (arowla) [inline webservices/resources/reports.py]: Looks like this got cherry-picked in here? Or needs a merge?

Comment (jmcarp) [inline webservices/resources/reports.py]: Was reconciling branches and forgot to take debugger out here. This is fixed, and I just merged #642 here to deal with merge conflicts.

**Comment by arowla** [comment]
Was this intended to also contain the `idx` primary key fixes? Having a little trouble following what's going on here.

**Comment by jmcarp** [comment]
@arowla I did merge in the primary key fix because the two branches kept diverging and giving me merge headaches. I'll separate if it would be better for review.

**Comment by jmcarp** [comment]
Replaced by #651.

**Pull request closed without merging**